contracts/apps/AppStorage.sol

```diff
@@ -4,14 +4,14 @@
 
 pragma solidity ^0.4.18;
 
-import "../kernel/IKernel.sol";
+import "../common/UnstructuredStorage.sol";
 
 
-contract AppStorage {
-    IKernel public kernel;
-    bytes32 public appId;
-    address internal pinnedCode; // used by Proxy Pinned
-    uint256 internal initializationBlock; // used by Initializable
-    uint256[95] private storageOffset; // forces App storage to start at after 100 slots
-    uint256 private offset;
+contract AppStorage is UnstructuredStorage {
+    // keccak256("IKernel.kernel")
+    bytes32 public constant kernelPosition = 0xe30296d9191ef86f01c8532636453e486db506a7be081a32b88c263fbc3a5e15;
+    // keccak256("bytes32.appIdPosition")
+    bytes32 public constant appIdPosition = 0x71ca50ba614d1d1a9a2433a5d0187a13e1a8c20bbe0013968dd089c71fd760eb;
+    // keccak256("address.pinnedCode"), used by Proxy Pinned
+    bytes32 public constant pinnedCodePosition = 0x0e2081dcdbb92be1037e55b39e913e70697c33a8a8b9bb01110e6f9e576089b8;
```

Comment:
I think the storage positions are an implementation detail that doesn't need to be exposed in the public API of the contract.

For all values I would add a getter and setter so the fact that AppStorage is using UnstructuredStorage is abstracted away from people interacting with the contract or deriving it:

```
function kernel() public view returns (IKernel) {
  return IKernel(getStorageAddress(kernelPosition));
}

function setKernel(IKernel newKernel) internal {
  setStorageAddress(kernelPosition, address(_kernel));
}
```